doc/user/client.md

```diff
@@ -0,0 +1,259 @@
+# Operator-SDK: Controller Runtime Client API
+
+## Overview
+
+The [kubernetes-sigs][org-kubernetes-sigs] organization builds tools to call the [Kubernetes][site-kubernetes] API in a clean, abstracted manner, much like the Operator SDK. The [controller-runtime][repo-controller-runtime] project is meant to help users build [controllers][site-kubernetes-controllers] easily by generating a controller project that interacts with a k8s cluster via CRUD operations. User-defined controller can perform specific tasks in a cluster; [operators][site-operators] can use multiple controllers to perform a variety of tasks. As operators use at least one controller, the SDK can rely on controller-runtime's k8s API code rather than develop a parallel set of API calls to execute the same cluster operations, namely CRUD operations.
+
+controller-runtime defines several interfaces used for cluster interaction:
+- `client.Client`: implementers perform CRUD operations on a k8s cluster.
+- `manager.Manager`: manages shared dependencies, such as Caches and Clients.
+- `reconcile.Reconciler`: compares provided state with actual cluster state and updates the cluster on finding state differences using a Client.
+
+The SDK relies on a `manager.Manager` to create a split `client.Client` interface that performs Create, Update, Delete, Get, and List operations within a `reconcile.Reconciler`'s Reconcile function.
+
+## The default Client
+
+The SDK will generate code creating a Manager. Managers hold a Cache and a Client that are used in CRUD operations and interface with the API server. The particular Client the SDK uses by default is referred to as a [split client][doc-split-client], created when instantiating a Manager. A split client reads (Get and List) from the Cache and writes (Create, Update, Delete) to the API server. Reading from the Cache significantly reduces request load on the API server; as long as the Cache is updated by the API server, read operations are eventually consistent. 
+
+### Non-default Client
+
+An operator developer may wish to create their own Client with different read/write optimzations, for example. controller-runtime provides a [constructor][doc-client-constr] for Clients:
+
+```Go
+// New returns a new Client using the provided config and Options.
+func New(config *rest.Config, options client.Options) (client.Client, error)
+```
+
+The Manager's client field must be set with the new Client:
+
+```Go
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	
+	...
+)
+
+func main() {
+	...
+
+	cfg, _ := config.GetConfig()
+	mgr, _ := manager.New(cfg, manager.Options{Namespace: namespace})
```

Comment:
You shouldn't need to ever change the manager's client.

This section should just demonstrate how to create a new client object if you're not using the Reconciler's client for some reason. (Say in a utility function somewhere).
```Go
import (
    "sigs.k8s.io/controller-runtime/pkg/client"
)

cfg, _ := config.GetConfig()
client, _ := client.New(cfg, client.Options{})
```

Reply:
So instead of calling
```Go
func (r *ReconcileKind) Reconcile(...) {
    r.client.Get(...)
}
```

You would create a new client and use it instead:
```Go
func (r *ReconcileKind) Reconcile(...) {
    cfg, _ := config.GetConfig()
    client, _ := client.New(cfg, client.Options{})
    client.Get(...)
}
```

Reply:
Kind of. Although you shouldn't really be creating a new client in the reconcile loop.
That should be done in the constructor for your reconciler:
```Go
func newReconciler(mgr manager.Manager) reconcile.Reconciler {
        cfg, _ := config.GetConfig()
        client, _ := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
	return &ReconcileMemcached{client: client, scheme: mgr.GetScheme()}
}

type ReconcileMemcached struct {
	client client.Client
	scheme *runtime.Scheme
}
```
But even creating a new client in your reconciler isn't advised unless you know what you're doing. 
You'd need to set the scheme correctly as well so that client recognizes your custom types.

That's why I didn't specify where to create a new client so users don't get the idea to create a new client in their reconciler.
This is more for if you're writing a test and you need to instantiate a client for that. Or in some utility package.

Reply:
I'll remove the specifics.